### PR TITLE
Adding Collection Search Extension

### PR DESCRIPTION
**Pull request by tjellicoe-tpzuk**
**Related Issue(s):** #615 

**Description:** Adding Collection Search Extension as defined [here](https://github.com/stac-api-extensions/collection-search)

**PR Checklist:**

- [ ] `pre-commit` hooks pass locally
- [ ] Tests pass (run `make test`)
- [ ] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

File: stac_fastapi/extensions/stac_fastapi/extensions/core/collectionSearch/collectionSearch.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    SIMPLE_QUERY = "http://www.opengis.net/spec/ogcapi-common-2/1.0/conf/simple-query"
+
+@attr.s
+class CollectionSearchExtension(ApiExtension):
+    """CollectionSearch Extension.
+
+    The collection search extension adds two endpoints which allow searching of 
+    collections via GET and POST:
+        GET /collection-search
+        POST /collection-search
```
Comment (vincentsarago) [inline stac_fastapi/extensions/stac_fastapi/extensions/core/collectionSearch/collectionSearch.py:37]: we should add a note that the extension seems to mention `GET /collections` and `POST /collections` but due to possible conflict with the `transaction` extension we should to use `/collection-search`

also wonder if we should use `/collections-search` or `/search/collections` 🤷‍♂️

File: stac_fastapi/api/stac_fastapi/api/app.py [hunk 2]
```diff
@@ -175,7 +177,7 @@ def register_get_item(self):
         """
         self.router.add_api_route(
             name="Get Item",
-            path="/collections/{collection_id}/items/{item_id}",
+            path="/catalogs/{catalog_id}collections/{collection_id}/items/{item_id}",
```
Comment (vincentsarago) [inline stac_fastapi/api/stac_fastapi/api/app.py:180]: 😬 maybe this change wasn't made on purpose?

**Review by m-mohr — CHANGES_REQUESTED** [review]
This doesn't implement https://github.com/stac-api-extensions/collection-search

The extension doesn't define a POST /collections endpoint, only GET /collections. The endpoint names are wrong.

> All functionality in OGC API - Records - Part 1: Local Resource Catalogue is only defined for the GET method (i.e. GET /collections). Note: STAC may add behavior for POST /collections in the future, but due to a potential conflict with the Transaction Extension, specific rules for content negotiation might be required.

**Comment by tjellicoe-tpzuk** [comment]
Thanks for your comments @vincentsarago and @m-mohr, this PR has become mixed up with subsequent commits, I will raise a new one only containing the updates for the collection search extension - see  draft PR #693

**Pull request closed without merging**